applications/StructuralMechanicsApplication/custom_utilities/tangent_operator_calculator_utility.h

```diff
@@ -288,8 +292,8 @@ class TangentOperatorCalculatorUtility
         Matrix& rPerturbedDeformationGradient,
         const Matrix& rDeformationGradientGP,
         const double Perturbation,
-        const std::size_t ComponentI,
-        const std::size_t ComponentJ
+        const SizeType ComponentI,
+        const SizeType ComponentJ
         )
     {
         rPerturbedDeformationGradient = rDeformationGradientGP;
```

Comment:
i am not convinced that doing it this way is the correct way.

recall that the perturbation dF is related to the strain increment de. If you prescribed de than you also prescribed dF (meaning for example that you should perturb only 6 values instead of 9, in the 3D case)

Reply:
I cannot comment on line 256, however your implementation is extremelty inefficient, since you compute minimal values PER EACH COEFFICIENT, which also internally uses an allocation!! same for line 252

Reply:
The deformation gradient is not symmetric, you need to perturbate the nine ones

Reply:
I am aware it is non symmetric. nevertheless the relevant change is in Ftrans*F, which is what enters in the strain computation. I think you can effectively have  perturbations instead of 9, and still get the same result. That would improve pretty sensitively the efficiency

Reply:
Does it work :+1: